integration/tests/identity_provider/identity_provider_test.go

```diff
@@ -0,0 +1,379 @@
+				"key": "value",
+			},
+		}, metav1.CreateOptions{})
+
+		Expect(err).To(HaveOccurred())
+
+		By("disassociating the identity provider")
+		cmd = params.EksctlCmd.
+			WithArgs(
+				"disassociate",
+				"identityprovider",
+				"--config-file", "-",
+				"--wait",
+			).
+			WithStdin(strings.NewReader(identityProviderClusterConfig)).
+			WithoutArg("--region", params.Region).
+			WithTimeout(45 * time.Minute)
+
+		Expect(cmd).To(RunSuccessfully())
+
+	})
+
+})
+
+var _ = AfterSuite(func() {
+	if !params.SkipCreate && !params.SkipDelete {
+		params.DeleteClusters()
+	}
+	if cleanupCognitoResources != nil {
+		Expect(cleanupCognitoResources()).To(Succeed())
+	}
+})
+
+func createOIDCClientset(eksAPI eksiface.EKSAPI, o *OIDCConfig, clusterName string) (kubernetes.Interface, error) {
+	contextName := fmt.Sprintf("%s@%s", "test", clusterName)
+
+	cluster, err := eksAPI.DescribeCluster(&eks.DescribeClusterInput{
+		Name: aws.String(clusterName),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "describing cluster")
+	}
+
+	certData, err := base64.StdEncoding.DecodeString(*cluster.Cluster.CertificateAuthority.Data)
+	if err != nil {
+		return nil, errors.Wrap(err, "unexpected error decoding certificate authority data")
+	}
+
+	config := clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			clusterName: {
+				Server:                   *cluster.Cluster.Endpoint,
+				CertificateAuthorityData: certData,
+			},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			contextName: {
+				Cluster:  clusterName,
+				AuthInfo: contextName,
+			},
+		},
+		AuthInfos: map[string]*clientcmdapi.AuthInfo{
+			contextName: {
+				AuthProvider: &clientcmdapi.AuthProviderConfig{
+					Name: "oidc",
+					Config: map[string]string{
+						"client-id":      o.clientID,
+						"id-token":       o.idToken,
+						"refresh-token":  o.refreshToken,
+						"idp-issuer-url": o.idpIssuerURL,
+					},
+				},
+			},
+		},
+		CurrentContext: contextName,
+	}
+
+	clientConfig, err := clientcmd.NewDefaultClientConfig(config, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "creating default client config")
+	}
+
+	return kubernetes.NewForConfig(clientConfig)
+}
+
+func setupCognitoProvider(clusterName, region string) (*OIDCConfig, error) {
+	c := cognitoidentityprovider.New(NewSession(region))
+
+	pool, err := c.CreateUserPool(&cognitoidentityprovider.CreateUserPoolInput{
+		Policies: &cognitoidentityprovider.UserPoolPolicyType{
+			PasswordPolicy: &cognitoidentityprovider.PasswordPolicyType{
+				MinimumLength:    aws.Int64(10),
+				RequireLowercase: aws.Bool(false),
+				RequireNumbers:   aws.Bool(true),
+				RequireSymbols:   aws.Bool(true),
+				RequireUppercase: aws.Bool(false),
+			},
+		},
+		PoolName:           aws.String(clusterName),
+		UsernameAttributes: aws.StringSlice([]string{"email"}),
+	})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "creating user pool")
```

Comment:
errors.Wrap(err

should use fmt.Errorf for this now everywhere, right?

Reply:
+1